src/IO/LimitReadBuffer.cpp

```diff
@@ -29,7 +29,9 @@ bool LimitReadBuffer::nextImpl()
 
     if (!in->next())
     {
+        /// Clearing the buffer with existing data.
         working_buffer = in->buffer();
+        pos = working_buffer.begin();
```

Comment:
Because of this - https://github.com/ClickHouse/ClickHouse/blob/b24d57db1cca93babf980b239a745bd7de7fa8e2/src/IO/LimitReadBuffer.cpp#L20

So once next() will return false here it will be leaved in the incorrect state, and subsequent next is UB.

Reply:
Ok, understood. Though it seems it's better to write something like `set(in->position(), 0)` here instead of these two lines. Using of `in->buffer().begin()` to set the current position can sometimes be wrong (see `nextimpl_working_buffer_offset` in `ReadBuffer`).

Reply:
>Using of in->buffer().begin() to set the current position can sometimes be wrong

Nice catch, thanks, fixed.